.github/workflows/release_ruleset.yaml

```diff
@@ -2,28 +2,47 @@
 # See example usage in https://github.com/bazel-contrib/rules-template/blob/main/.github/workflows/release.yaml
 #
 # By default this workflows calls `.github/workflows/release_prep.sh` as the command to prepare
-# the release. This can be customized with the `release_prep_command` attribute. Release notes are
-# expected to be outputted to stdout from the release prep command.
+# the release. This can be customized with the `release_prep_command` attribute. See documentation
+# for the `release_prep_command` property below for inputs and outputs of the script.
 #
 # This workflow uses https://github.com/bazel-contrib/setup-bazel to prepare the cache folders.
 # Caching may be disabled by setting `mount_bazel_caches` to false.
+#
+# The workflow requires the following permissions to be set on the invoking job:
+#
+# permissions:
+#   contents: write        # Needed for uploading release files
+
 
 on:
   # Make this workflow reusable, see
   # https://github.blog/2022-02-10-using-reusable-workflows-github-actions
   workflow_call:
     inputs:
-      release_files:
-        required: true
-        description: |
-          Newline-delimited globs of paths to assets to upload for release.
-          See https://github.com/softprops/action-gh-release#inputs
-        type: string
```

Comment:
The files to upload to the release are now determined by the `release_prep.sh` script. The script is supplied with a pre-created directory defined in `$ARTIFACTS_DIR` to put all artifacts. The script also returns a JSON blob pointing out which artifact(s) are release archives so that we know what to attest.

A couple of alternative solutions are:
* We keep `release_files` but also add `attest_files`.
* We automatically attest all files under `$ARTIFACTS_DIR`, not just the release archives.

Reply:
In my opinion, I like giving all of the responsibility for the release to the `release_prep.sh` script, rather than partially the script and partially some inputs.